dataprofiler/data_readers/json_data.py

```diff
@@ -51,83 +56,201 @@ def __init__(self, input_file_path=None, data=None, options=None):
+                    list_of_dict = list_of_dict + self._find_data(json_data[key], path + key)
+                else:
+                    list_of_dict = list_of_dict + [{path + key: json_data[key]}]
+        elif isinstance(json_data, list):
+            if all(isinstance(x, dict) for x in json_data):
+                for key in json_data:
+                    list_of_dict = list_of_dict + self._find_data(key, path)
+            else:
+                list_of_dict = list_of_dict + [{path[:-len(self._key_separator)]: json_data}]
+        else:
+            list_of_dict = list_of_dict + [{path[:-len(self._key_separator)]: json_data}]
+        return list_of_dict
+    
+    def _coalesce_dicts(self, list_of_dicts):
+        """
+        Merges all the dictionaries into as few dictionaries as possible.
+
+        :param list_of_dicts: the list of dictionaries to be coalesced
+        :type list_of_dicts: list(dict)
+        :return: Coalesced list of dictionaries
+        """
+        coalesced_list_of_dicts = [{}]
+        for item in list_of_dicts:
+            found = False
+            for dict_items in coalesced_list_of_dicts:
+                if list(item.keys())[0] not in dict_items:
+                    dict_items.update(item)
+                    found = True
+                    break
+            if not found:
+                coalesced_list_of_dicts.append(item)
+        return coalesced_list_of_dicts
+
+    def _get_data_as_flattened_dataframe(self, json_lines):
+        """
+        Loads the data when in a JSON format from a data stream (nested
+        lists of dictionaries and a key value for a payload.
+
         :param json_lines: json format list of dicts or dict
         :return:
         """
+        if isinstance(json_lines, pd.DataFrame):
+            return json_lines
+        payload_data = None
         if isinstance(json_lines, dict):
-            json_lines = [json_lines]
+            # Glean Payload Data
+            found_payload_key = None
+            for payload_key in self._payload_keys:
+                if payload_key in json_lines.keys():
+                    payload_data = json_lines[payload_key]
+                    if isinstance(payload_data, dict):
+                        payload_data = self._find_data(payload_data)
+                        payload_data = self._coalesce_dicts(payload_data)
+                    payload_data, original_df_dtypes = data_utils.json_to_dataframe(
+                        json_lines=payload_data,
+                        selected_columns=self.selected_keys,
+                        read_in_string=False
+                    )
+                    for column in payload_data.columns:
+                        payload_data.rename(
+                            columns={column: payload_key + self._key_separator + str(column)},
+                            inplace=True)
+                    found_payload_key = payload_key
+                    break
```

Comment:
I see the break on if a payload is found, does the other payload (if two are found) become metadata. Could select largest one?

Reply:
I guess most rows 🤷‍♂️, if they match they should probably create a single dataset?